.github/workflows/mac_ci.yml

```
@@ -0,0 +1,70 @@
+name: MacRelease
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master and release brnaches
+on:
+  push:
+    branches: [ master, rel-* ]
+  # adding for testing
+  pull_request:
+    branches: [ master, rel-* ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: macos-11.0
+    strategy:
+      matrix:
+        python-version: [3.9]
+        architecture: ['x86_64']
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        
+      - name: Checkout submodules
+        shell: bash
+        run: |
+          auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+          git submodule sync --recursive
+          git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+          
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
```

```suggestion
        uses: actions/setup-python@v4
```

looks like the latest version is v4. was there a particular reason to use v1?